core/kazoo_numbers/src/knm_port_request.erl

```diff
@@ -175,6 +177,27 @@ account_active_ports(AccountId) ->
             {'error', 'not_found'}
     end.
 
+%%------------------------------------------------------------------------------
+%% @doc
+%% @end
+%%------------------------------------------------------------------------------
+-spec account_ports_by_state(kz_term:ne_binary(), kz_term:ne_binary()) -> {'ok', kz_json:objects()} |
+          {'error', 'not_found'}.
+account_ports_by_state(AccountId, PortState) ->
+    ViewOptions = [{'startkey', [AccountId, PortState]}
+                  ,{endkey, [AccountId, PortState, kz_json:new()]}
+                  ,'include_docs'
+                  ],
+    case lists:member(PortState, ?PORT_STATES) %% Make sure it is a valid state
```

Comment:
This is an internal function, no need for checking state here.

But if you do want to check anyway this could result in `false` case which is going to crash, because there is no case clause for it below.

Also please don't abuse logical expression, you're mixing a boolean operation by non boolean operation. Though it is valid syntax, but it is actually a side effect of the implementation.